client/src/partials/stock/donation_management/donation_management.js

```diff
@@ -429,49 +422,54 @@ function DonationManagementController($scope, $q, $translate, $location, $routeP
     session.configured = false;
   }
 
-  function simulatePurchase() {
-    if (session.donation.items.length > 0) {
+  function handleError(error) {
+    $translate('PURCHASE.WRITE_FAILED')
+    .then(function (value) {
+       messenger.danger(value);
+    });
+  }
+  // End modules functions
 
-      var purchase = {
-        uuid          : uuid(),
-        cost          : simulatePurchaseTotal(),
-        purchase_date : util.sqlDate(session.config.date),
-        currency_id   : $scope.enterprise.currency_id,
-        creditor_uuid : null,
-        purchaser_id  : session.config.employee.id, //$scope.user.data.id,
-        project_id    : $scope.project.id,
-        note          : 'DONATION ' + session.config.donor.name + '/' + util.sqlDate(session.config.date),
-        receiver_id   : session.config.employee.id,
-        paid          : 0,
-        confirmed     : 0,
-        closed        : 0,
-        is_donation   : 1,
-        is_direct     : 0
-      };
+  // Simulations functions
+  function simulatePurchase() {
+    if (session.donation.items.length <= 0) { throw new Error('Need donation items'); }
+
+    var def = $q.defer();
+
+    var purchase = {
+      uuid          : uuid(),
+      cost          : simulatePurchaseTotal(),
+      purchase_date : util.sqlDate(session.config.date),
+      currency_id   : $scope.enterprise.currency_id,
+      creditor_uuid : null,
+      purchaser_id  : session.config.employee.id, //$scope.user.data.id,
+      project_id    : $scope.project.id,
+      note          : 'DONATION ' + session.config.donor.name + '/' + util.sqlDate(session.config.date),
+      receiver_id   : session.config.employee.id,
+      paid          : 0,
+      confirmed     : 0,
+      closed        : 0,
+      is_donation   : 1,
+      is_direct     : 0
+    };
 
-      simulateWritePurchaseLine(purchase)
-      .then(simulateWritePurchaseItems(purchase.uuid))
-      .then(updateStockPurchaseOrder(purchase.uuid))
-      .catch(handleError);
-    }
+    simulateWritePurchaseLine(purchase)
```

Comment:
This can be `return simulateWritePurchaseLine(purchase)` and you can avoid using a `defer()`.